public/app/plugins/datasource/tempo/NativeSearch.tsx

```diff
@@ -48,14 +52,24 @@ const NativeSearch = ({ datasource, query, onChange, onBlur, onRunQuery }: Props
     serviceNameOptions: [],
     spanNameOptions: [],
   });
+  const [error, setError] = useState(null);
 
   useEffect(() => {
     const fetchAutocomplete = async () => {
-      await languageProvider.start();
-      const serviceNameOptions = await languageProvider.getOptions('service.name');
-      const spanNameOptions = await languageProvider.getOptions('name');
-      setHasSyntaxLoaded(true);
-      setAutocomplete({ serviceNameOptions, spanNameOptions });
+      try {
+        await languageProvider.start();
+        const serviceNameOptions = await languageProvider.getOptions('service.name');
+        const spanNameOptions = await languageProvider.getOptions('name');
+        setHasSyntaxLoaded(true);
+        setAutocomplete({ serviceNameOptions, spanNameOptions });
+      } catch (error) {
+        // Display message if Tempo is connected but search 404's
+        if (error?.status === 404) {
```

Comment:
The error message is very generic saying "Unable to connect to Tempo". I think it might be confusing because Tempo is up and running, it's just the search is not available. Could we make it more specific that Grafana was unable to connect to Tempo _Search_?

Reply:
Good point, I'll make that change.